_posts/2021-03-15-changelog-143.markdown

```diff
@@ -0,0 +1,92 @@
+---
+layout: post
+title: "IntelliJ Rust Changelog #143"
+date: 2021-03-15 11:00:00 +0300
+---
+
+## New Features
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6564 -->
+* Provide initial support for [function-like](https://doc.rust-lang.org/reference/procedural-macros.html#function-like-procedural-macros) procedural macros.
+  Now the plugin can expand such procedural macro calls and as a result they
+  automatically get a bunch of some features available for declarative macros:
+  highlighting, name resolution, restricted completion, `Show macro expansion` intentions and others.  
+  Note, it's only initial implementation, so it may work in unexpected way in some cases.
+  The feature is disabled for default for now, to turn if on,
+  you should enable `org.rust.cargo.evaluate.build.scripts` and `org.rust.macros.proc` experimental features.
+  Don't forget to reload project structure after enabling the corresponding features via `Refresh Cargo Projects` action on Cargo tool window.  
+  See [tracking issue](https://github.com/intellij-rust/intellij-rust/issues/6908) for more
+  details and current status of procedural macros support
+
+  {% include gif-img.html path="/assets/posts/changelog-143/function-like-proc-macro-support" w="700px" %}
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6933 -->
+* Auto-indent line when typing <kbd>Enter</kbd> inside an array literal
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6901 -->
+* TOML syntax highlighting is now available in `Cargo.toml.orig` files
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6941 -->
+* Provide new plugin icon
+
+## Performance Improvements
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6917 -->
+* Prevent a possible deadlock during project structure setup
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6861 -->
+* Slightly optimize name resolution
+  with [new name resolution](https://github.com/intellij-rust/intellij-rust/issues/6217) engine
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6899 -->
+* Exclude more stdlib parts from indexing. Slightly speed up index process
+
+## Fixes
+
+* [New name resolution](https://github.com/intellij-rust/intellij-rust/issues/6217) engine improvements:
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/6872 -->
+    * Find all possible impls for type aliases regardless type aliases number for particular type.
```

Comment:
```suggestion
    * Find all possible impls for a type alias regardless of the number of aliases defined for the corresponding type.
```
BTW not sure if should we use "impls" or "`impl`s" (there are lots of usages of both in our changelogs)

Reply:
I think `impl` block is more correct